### DISCUSSION TITLE
Supabase Database Indexing: Does Supabase automatically create database indexes?

### DISCUSSION THREAD

**Discussion by aidrecabrera** [supabase/supabase#21672] [Questions] [answered]
Hello there,

I've been trying to explore the Supabase's approach to handling database indexes.

**Question:** Does Supabase automatically create database indexes when I create the tables?

I am curious whether it necessitates manual intervention or the index creation is already taken care of? 

I have discovered the `"Database Indexes"` section, and there are already queries for creating the indexes. So is it safe to assume that Supabase does implement the indexes automatically?

Another thing, while querying my database using the `JavaScript Client Library`, am I already benefiting from the advantages of these indexes? or if there are additional setup steps required within my project to fully leverage and optimize database indexing?

Your insights would be greatly appreciated. Thank you in advance.

**Comment by GaryAustin1** [comment, marked as answer]
No.  The only indexes created are those done by Postgres.  That includes primary key and unique constraint.

After you create indexes (or they are created in the two above cases) then nothing else needs to be done for API use.

**Reply by jtomaszewski** [reply]
This is so surprising! In most frameworks creating an index (e.g. on a foreign key column) is an integral part of creating a table (or adding a column). Yet Supabase UI for editing tables doesn't allow for that. Indexes are "hidden" in a separate view.

Because of that (lack of "enable/disable" index in the edit table UI), I was pretty sure Supabase does create indexes automatically for my foreign keys... only after delivering the app to production I've learned that I have literally zero indexes on my database (besides the primary keys). Now I have to 1-by-1 go through all of my tables and add all missing indexes.

(Index advisor doesn't suffice me. I need to predict indexes before they're needed, not after.)
